packages/angular_devkit/build_angular/src/tools/esbuild/application-code-bundle.ts

```diff
@@ -245,6 +283,19 @@ function getEsBuildCommonOptions(options: NormalizedApplicationBuildOptions): Bu
     jit,
   } = options;
 
+  // Ensure unique hashes for i18n translation changes when using post-process inlining.
```

Comment:
I am assuming that this is done so that the hash in the filename is also updated. (Maybe add this to the comment to make it more clear of the intention to force a unique hash).

Is there any other reason?

Reply:
That's correct. There's custom code in the Webpack builder to do something similar.
I'll add a comment.